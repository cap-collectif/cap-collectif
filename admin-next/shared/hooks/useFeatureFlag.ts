import { useLazyLoadQuery, graphql } from 'react-relay'
import invariant from '../utils/invariant'
import { useFeatureFlagQuery, FeatureFlagType } from '@relay/useFeatureFlagQuery.graphql'

type FeatureFlags = Omit<
  {
    [key in FeatureFlagType]: boolean
  },
  '%future added value'
>

const query = graphql`
  query useFeatureFlagQuery {
    featureFlags {
      type
      enabled
    }
  }
`
export const defaultFeatureFlags: FeatureFlags = {
  report_browers_errors_to_sentry: false,
  login_saml: false,
  oauth2_switch_user: false,
  votes_min: false,
  blog: false,
  calendar: false,
  login_facebook: false,
  privacy_policy: false,
  members_list: false,
  captcha: false,
  turnstile_captcha: false,
  questionnaire_result: false,
  consent_external_communication: false,
  consent_internal_communication: false,
  newsletter: false,
  profiles: false,
  projects_form: false,
  project_trash: false,
  search: false,
  share_buttons: false,
  shield_mode: false,
  registration: false,
  phone_confirmation: false,
  reporting: false,
  themes: false,
  districts: false,
  user_type: false,
  votes_evolution: false,
  restrict_registration_via_email_domain: false,
  export: false,
  server_side_rendering: false,
  zipcode_at_register: false,
  consultation_plan: false,
  display_map: false,
  sso_by_pass_auth: false,
  allow_users_to_propose_events: false,
  secure_password: false,
  restrict_connection: false,
  login_franceconnect: false,
  read_more: false,
  display_pictures_in_depository_proposals_list: false,
  external_project: false,
  multilangue: false,
  display_pictures_in_event_list: false,
  emailing: false,
  emailing_parameters: false,
  emailing_group: false,
  proposal_revisions: false,
  import_proposals: false,
  analytics_page: false,
  http_redirects: false,
  project_admin: false,
  developer_documentation: false,
  export_legacy_users: false,
  graphql_query_analytics: false,
  noindex_on_profiles: false,
  new_project_card: false,
  indexation: false,
  twilio: false,
  remind_user_account_confirmation: false,
  anonymous_questionnaire: false,
  login_cas: false,
  login_openid: false,
  public_api: false,
  sentry_log: false,
  versions: false,
  paper_vote: false,
  graphql_introspection: false,
  helpscout_beacon: false,
  api_sendinblue: false,
  proposal_sms_vote: false,
  organizations: false,
  moderation_comment: false,
  unstable__new_create_project: false,
  new_vote_step: false,
  mediator: false,
  online_help: false,
  full_proposal_card: false,
  multi_consultations: false,
}

type FeatureFlagsEnabled<T> = {
  [key in T as FeatureFlagType]?: boolean
}

export const useFeatureFlag = (flag: FeatureFlagType): boolean => {
  const data = useLazyLoadQuery<useFeatureFlagQuery>(query, {}, { fetchPolicy: 'store-only' })
  invariant(data.featureFlags !== undefined, 'The featureFlags are missing in Relay store.')

  const featureFlag = data.featureFlags.find(f => f.type === flag)
  invariant(featureFlag !== undefined, 'The featureFlag "%s" is missing in Relay store.', flag)
  return featureFlag ? featureFlag.enabled : false
}

export const useFeatureFlags = <T extends FeatureFlagType>(flags: FeatureFlagType[]): FeatureFlagsEnabled<T> => {
  const data = useLazyLoadQuery<useFeatureFlagQuery>(query, {}, { fetchPolicy: 'store-only' })
  invariant(data.featureFlags !== undefined, 'The featureFlags are missing in Relay store.')

  return flags.reduce<FeatureFlagsEnabled<T>>((acc, featureFlag) => {
    const currentFlag = data.featureFlags.find(f => f.type === featureFlag)

    if (currentFlag) {
      acc[featureFlag] = currentFlag.enabled
    } else {
      invariant(currentFlag !== undefined, 'The featureFlag "%s" is missing in Relay store.', featureFlag)
    }

    return acc
  }, {})
}

export const useAllFeatureFlags = (): FeatureFlags => {
  const data = useLazyLoadQuery<useFeatureFlagQuery>(query, {}, { fetchPolicy: 'store-only' })
  invariant(data.featureFlags !== undefined, 'The featureFlags are missing in Relay store.')

  return data.featureFlags.reduce<FeatureFlags>((acc, featureFlag) => {
    acc[featureFlag.type] = featureFlag.enabled
    return acc
  }, defaultFeatureFlags)
}

export default useFeatureFlag
