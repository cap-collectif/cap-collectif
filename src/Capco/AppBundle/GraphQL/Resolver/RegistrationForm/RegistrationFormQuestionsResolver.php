<?php

namespace Capco\AppBundle\GraphQL\Resolver\RegistrationForm;

use Capco\AppBundle\Entity\RegistrationForm;
use Overblog\GraphQLBundle\Definition\Resolver\QueryInterface;

class RegistrationFormQuestionsResolver implements QueryInterface
{
    public function __invoke(RegistrationForm $form): array
    {
        $questions = $form->getRealQuestions()->toArray();
        usort(
            $questions,
            fn ($a, $b) => $a->getQuestionnaireAbstractQuestion()->getPosition() <=>
            $b->getQuestionnaireAbstractQuestion()->getPosition()
        );

        return $questions;
    }
}
