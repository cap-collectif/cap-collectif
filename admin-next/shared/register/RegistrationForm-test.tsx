/* eslint-env jest */
import { graphql, useLazyLoadQuery } from 'react-relay'
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils'
import { addsSupportForPortals, RelaySuspensFragmentTest, FormWrapper } from 'tests/testUtils'
import type { RegistrationFormTestQuery } from '@relay/RegistrationFormTestQuery.graphql'
import { RegistrationForm } from './RegistrationForm'
import useFeatureFlag, { useFeatureFlags } from '@shared/hooks/useFeatureFlag'
import { render } from '@testing-library/react'

jest.mock('@shared/hooks/useFeatureFlag')

describe('<RegistrationForm />', () => {
  let environment
  let TestComponent

  const query = graphql`
    query RegistrationFormTestQuery @relay_test_operation {
      ...RegistrationForm_query
    }
  `

  const mock = {
    Query: () => ({
      organizationName: { value: 'Mon site' },
      internalCommunicationFrom: { value: 'Cap collectif' },
      userTypes: [
        {
          value: '1',
          label: 'Association',
        },
        {
          value: '2',
          label: 'Citoyen',
        },
      ],
    }),
  }

  beforeEach(() => {
    addsSupportForPortals()
    environment = createMockEnvironment()

    const TestRenderer = props => {
      const data = useLazyLoadQuery<RegistrationFormTestQuery>(query, {})
      if (!data) return null
      return <RegistrationForm query={data} {...props} />
    }

    TestComponent = props => (
      <RelaySuspensFragmentTest environment={environment}>
        <FormWrapper>
          <TestRenderer {...props} />
        </FormWrapper>
      </RelaySuspensFragmentTest>
    )
  })
  it('renders a form with inputs and a captcha', () => {
    // @ts-ignore jest
    useFeatureFlag.mockImplementation(() => false)
    // @ts-ignore jest
    useFeatureFlags.mockImplementation(() => {
      return { captcha: true }
    })
    environment.mock.queueOperationResolver(operation => MockPayloadGenerator.generate(operation, mock))
    const { asFragment } = render(<TestComponent />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders a form with user_type enabled', () => {
    // @ts-ignore jest
    useFeatureFlag.mockImplementation(() => false)
    // @ts-ignore jest
    useFeatureFlags.mockImplementation(() => {
      return { user_type: true }
    })
    environment.mock.queueOperationResolver(operation => MockPayloadGenerator.generate(operation, mock))
    const { asFragment } = render(<TestComponent />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders a form with zipcode enabled', () => {
    // @ts-ignore jest
    useFeatureFlag.mockImplementation(() => false)
    // @ts-ignore jest
    useFeatureFlags.mockImplementation(() => {
      return { zipcode_at_register: true }
    })
    environment.mock.queueOperationResolver(operation => MockPayloadGenerator.generate(operation, mock))
    const { asFragment } = render(<TestComponent />)
    expect(asFragment()).toMatchSnapshot()
  })
})
