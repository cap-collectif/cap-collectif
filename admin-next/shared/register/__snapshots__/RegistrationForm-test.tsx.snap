// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationForm /> renders a form with inputs and a captcha 1`] = `
[
  <div
    className="sc-bcXGCL gHoLKQ toasts-container toasts-container--top"
  />,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor=":r0:"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        global.email
      </span>
    </label>
    <input
      aria-required={true}
      autoComplete="email"
      autoFocus={true}
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id=":r0:"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="email.placeholder"
      readOnly={false}
      required={true}
      type="email"
    />
  </div>,
  <div
    className="sc-bcXGCL klvDp"
  >
    <div
      className="sc-bcXGCL hQCGLB cap-form-control"
    >
      <label
        className="sc-bcXGCL fTPHCc cap-form-label"
        htmlFor=":r1:"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        <span
          className="sc-bcXGCL dQChsN cap-text"
          style={
            {
              "marginLeft": 0,
              "marginTop": 0,
            }
          }
        >
          registration.password
        </span>
      </label>
      <div
        className="sc-bcXGCL sc-dwnPIk hsfhbV eCNeCU cap-input-outer"
      >
        <input
          aria-required={true}
          autoComplete="off"
          autoFocus={true}
          className="sc-bcXGCL dANhtW cap-input"
          disabled={false}
          id=":r1:"
          name="plainPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          required={true}
          type="password"
        />
        <button
          aria-describedby=":r2:"
          className="sc-bcXGCL fROdoz"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onClickCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDownCapture={[Function]}
          onKeyPressCapture={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseMove={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="sc-bcXGCL sc-oZHua JBVPO cCzERh cap-icon cap-input-icon"
            fill="none"
            height="1em"
            viewBox="0 0 12 10"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1.903 9 8.333-8M4.65 8.134c.437.135.892.203 1.35.2 1.822.03 3.67-1.25 4.81-2.506a.74.74 0 0 0 0-.99 9 9 0 0 0-1.366-1.225M7.072 2.459a4.2 4.2 0 0 0-1.073-.126c-1.791-.03-3.644 1.223-4.81 2.505a.74.74 0 0 0 0 .99q.534.582 1.156 1.066"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={0.8}
            />
            <path
              d="M4.333 5.333a1.666 1.666 0 0 1 1.666-1.666M7.666 5.333A1.667 1.667 0 0 1 5.999 7"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={0.8}
            />
          </svg>
        </button>
        <span
          hidden={true}
          id=":r2:"
          style={
            {
              "position": "fixed",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor="username"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        global.fullname
      </span>
    </label>
    <input
      aria-required={true}
      autoComplete="username"
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id="username"
      minLength={2}
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="global.placeholder.name"
      readOnly={false}
      required={true}
      type="text"
    />
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor="responses.0.value"
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        &lt;mock-value-for-field-"title"&gt;
      </span>
      <p
        className="sc-bcXGCL gPctOm cap-text"
      >
        global.optional
      </p>
    </label>
    <span
      className="sc-bcXGCL ifYgMR cap-form-guideline"
    >
      &lt;mock-value-for-field-"helpText"&gt;
    </span>
    <div
      className=" ql-editor wysiwyg-render"
      dangerouslySetInnerHTML={
        {
          "__html": "<mock-value-for-field-"description">",
        }
      }
    />
    <input
      aria-required={false}
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id="responses.0.value"
      name="responses.0.value"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      type="text"
    />
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL hICbbA"
      htmlFor="charte"
    >
      <div
        className="sc-bcXGCL dawbeU cap-checkbox"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        <input
          aria-required={true}
          className="sc-bcXGCL leuqnG cap-checkbox__input"
          disabled={false}
          id="charte"
          name="charte"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          required={true}
          style={
            {
              "marginLeft": 0,
              "marginTop": 0,
            }
          }
          type="checkbox"
        />
        <div
          className="sc-bcXGCL cCxzTy cap-checkbox__box"
        />
      </div>
      <span
        className="sc-bcXGCL grQllb"
      >
        <span
          className="sc-bcXGCL sc-cc425a65-0 keKWCf hMXkYl"
        >
          registration.charte
        </span>
      </span>
    </label>
  </div>,
  <div
    id="recaptcha"
    style={
      {
        "transform": "scale(0.85)",
        "transformOrigin": "0 0",
      }
    }
  />,
]
`;

exports[`<RegistrationForm /> renders a form with user_type enabled 1`] = `
[
  <div
    className="sc-bcXGCL gHoLKQ toasts-container toasts-container--top"
  />,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor=":r8:"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        global.email
      </span>
    </label>
    <input
      aria-required={true}
      autoComplete="email"
      autoFocus={true}
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id=":r8:"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="email.placeholder"
      readOnly={false}
      required={true}
      type="email"
    />
  </div>,
  <div
    className="sc-bcXGCL klvDp"
  >
    <div
      className="sc-bcXGCL hQCGLB cap-form-control"
    >
      <label
        className="sc-bcXGCL fTPHCc cap-form-label"
        htmlFor=":r9:"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        <span
          className="sc-bcXGCL dQChsN cap-text"
          style={
            {
              "marginLeft": 0,
              "marginTop": 0,
            }
          }
        >
          registration.password
        </span>
      </label>
      <div
        className="sc-bcXGCL sc-dwnPIk hsfhbV eCNeCU cap-input-outer"
      >
        <input
          aria-required={true}
          autoComplete="off"
          autoFocus={true}
          className="sc-bcXGCL dANhtW cap-input"
          disabled={false}
          id=":r9:"
          name="plainPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          required={true}
          type="password"
        />
        <button
          aria-describedby=":ra:"
          className="sc-bcXGCL fROdoz"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onClickCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDownCapture={[Function]}
          onKeyPressCapture={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseMove={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="sc-bcXGCL sc-oZHua JBVPO cCzERh cap-icon cap-input-icon"
            fill="none"
            height="1em"
            viewBox="0 0 12 10"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1.903 9 8.333-8M4.65 8.134c.437.135.892.203 1.35.2 1.822.03 3.67-1.25 4.81-2.506a.74.74 0 0 0 0-.99 9 9 0 0 0-1.366-1.225M7.072 2.459a4.2 4.2 0 0 0-1.073-.126c-1.791-.03-3.644 1.223-4.81 2.505a.74.74 0 0 0 0 .99q.534.582 1.156 1.066"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={0.8}
            />
            <path
              d="M4.333 5.333a1.666 1.666 0 0 1 1.666-1.666M7.666 5.333A1.667 1.667 0 0 1 5.999 7"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={0.8}
            />
          </svg>
        </button>
        <span
          hidden={true}
          id=":ra:"
          style={
            {
              "position": "fixed",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor="username"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        global.fullname
      </span>
    </label>
    <input
      aria-required={true}
      autoComplete="username"
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id="username"
      minLength={2}
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="global.placeholder.name"
      readOnly={false}
      required={true}
      type="text"
    />
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor=":re:"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        registration.type
      </span>
      <p
        className="sc-bcXGCL gPctOm cap-text"
      >
        global.optional
      </p>
    </label>
    <div
      className="sc-bcXGCL gGWHIo"
    >
      <div
        className="cap-select css-b62m3t-container"
        id=":re:"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          className="cap-select__control css-1qmtci6-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="cap-select__value-container css-206zzd"
          >
            <div
              className="cap-select__placeholder css-rf6qli-placeholder"
              id="react-select-2-placeholder"
            >
              registration.type
            </div>
            <div
              className="cap-select__input-container css-1uvadrb"
              data-value=""
            >
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className="cap-select__input"
                disabled={false}
                id="userType"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <svg
            aria-hidden="true"
            className="sc-bcXGCL sc-oZHua dKAEWa cCzERh cap-icon"
            fill="none"
            height="1em"
            style={
              {
                "cursor": "pointer",
              }
            }
            viewBox="0 0 12 7"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 7c.164 0 .326-.032.475-.095q.227-.096.392-.27l4.916-5.216a.83.83 0 0 0 .214-.62.85.85 0 0 0-.311-.584.97.97 0 0 0-.66-.214.95.95 0 0 0-.632.277l-4.3 4.562a.133.133 0 0 1-.188 0L1.606.278A.95.95 0 0 0 .974.001a.97.97 0 0 0-.66.214A.85.85 0 0 0 .003.8a.83.83 0 0 0 .214.62l4.915 5.214c.109.115.242.208.392.27.15.064.312.097.476.097"
              fill="currentColor"
            />
          </svg>
        </div>
        <input
          name="userType"
          type="hidden"
          value=""
        />
      </div>
    </div>
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor="responses.0.value"
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        &lt;mock-value-for-field-"title"&gt;
      </span>
      <p
        className="sc-bcXGCL gPctOm cap-text"
      >
        global.optional
      </p>
    </label>
    <span
      className="sc-bcXGCL ifYgMR cap-form-guideline"
    >
      &lt;mock-value-for-field-"helpText"&gt;
    </span>
    <div
      className=" ql-editor wysiwyg-render"
      dangerouslySetInnerHTML={
        {
          "__html": "<mock-value-for-field-"description">",
        }
      }
    />
    <input
      aria-required={false}
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id="responses.0.value"
      name="responses.0.value"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      type="text"
    />
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL hICbbA"
      htmlFor="charte"
    >
      <div
        className="sc-bcXGCL dawbeU cap-checkbox"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        <input
          aria-required={true}
          className="sc-bcXGCL leuqnG cap-checkbox__input"
          disabled={false}
          id="charte"
          name="charte"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          required={true}
          style={
            {
              "marginLeft": 0,
              "marginTop": 0,
            }
          }
          type="checkbox"
        />
        <div
          className="sc-bcXGCL cCxzTy cap-checkbox__box"
        />
      </div>
      <span
        className="sc-bcXGCL grQllb"
      >
        <span
          className="sc-bcXGCL sc-cc425a65-0 keKWCf hMXkYl"
        >
          registration.charte
        </span>
      </span>
    </label>
  </div>,
]
`;

exports[`<RegistrationForm /> renders a form with zipcode enabled 1`] = `
[
  <div
    className="sc-bcXGCL gHoLKQ toasts-container toasts-container--top"
  />,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor=":rh:"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        global.email
      </span>
    </label>
    <input
      aria-required={true}
      autoComplete="email"
      autoFocus={true}
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id=":rh:"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="email.placeholder"
      readOnly={false}
      required={true}
      type="email"
    />
  </div>,
  <div
    className="sc-bcXGCL klvDp"
  >
    <div
      className="sc-bcXGCL hQCGLB cap-form-control"
    >
      <label
        className="sc-bcXGCL fTPHCc cap-form-label"
        htmlFor=":ri:"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        <span
          className="sc-bcXGCL dQChsN cap-text"
          style={
            {
              "marginLeft": 0,
              "marginTop": 0,
            }
          }
        >
          registration.password
        </span>
      </label>
      <div
        className="sc-bcXGCL sc-dwnPIk hsfhbV eCNeCU cap-input-outer"
      >
        <input
          aria-required={true}
          autoComplete="off"
          autoFocus={true}
          className="sc-bcXGCL dANhtW cap-input"
          disabled={false}
          id=":ri:"
          name="plainPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          required={true}
          type="password"
        />
        <button
          aria-describedby=":rj:"
          className="sc-bcXGCL fROdoz"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onClickCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDownCapture={[Function]}
          onKeyPressCapture={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          onMouseEnter={[Function]}
          onMouseMove={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="sc-bcXGCL sc-oZHua JBVPO cCzERh cap-icon cap-input-icon"
            fill="none"
            height="1em"
            viewBox="0 0 12 10"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1.903 9 8.333-8M4.65 8.134c.437.135.892.203 1.35.2 1.822.03 3.67-1.25 4.81-2.506a.74.74 0 0 0 0-.99 9 9 0 0 0-1.366-1.225M7.072 2.459a4.2 4.2 0 0 0-1.073-.126c-1.791-.03-3.644 1.223-4.81 2.505a.74.74 0 0 0 0 .99q.534.582 1.156 1.066"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={0.8}
            />
            <path
              d="M4.333 5.333a1.666 1.666 0 0 1 1.666-1.666M7.666 5.333A1.667 1.667 0 0 1 5.999 7"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={0.8}
            />
          </svg>
        </button>
        <span
          hidden={true}
          id=":rj:"
          style={
            {
              "position": "fixed",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor="username"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        global.fullname
      </span>
    </label>
    <input
      aria-required={true}
      autoComplete="username"
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id="username"
      minLength={2}
      name="username"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="global.placeholder.name"
      readOnly={false}
      required={true}
      type="text"
    />
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor=":rn:"
      style={
        {
          "marginLeft": 0,
          "marginTop": 0,
        }
      }
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        user.register.zipcode
      </span>
      <p
        className="sc-bcXGCL gPctOm cap-text"
      >
        global.optional
      </p>
    </label>
    <input
      autoComplete="postal-code"
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id=":rn:"
      name="zipcode"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      type="text"
    />
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL fTPHCc cap-form-label"
      htmlFor="responses.0.value"
    >
      <span
        className="sc-bcXGCL dQChsN cap-text"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        &lt;mock-value-for-field-"title"&gt;
      </span>
      <p
        className="sc-bcXGCL gPctOm cap-text"
      >
        global.optional
      </p>
    </label>
    <span
      className="sc-bcXGCL ifYgMR cap-form-guideline"
    >
      &lt;mock-value-for-field-"helpText"&gt;
    </span>
    <div
      className=" ql-editor wysiwyg-render"
      dangerouslySetInnerHTML={
        {
          "__html": "<mock-value-for-field-"description">",
        }
      }
    />
    <input
      aria-required={false}
      className="sc-bcXGCL sc-dwnPIk egzrup eCNeCU cap-input"
      disabled={false}
      id="responses.0.value"
      name="responses.0.value"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={false}
      required={false}
      type="text"
    />
  </div>,
  <div
    className="sc-bcXGCL fVgOgb cap-form-control"
  >
    <label
      className="sc-bcXGCL hICbbA"
      htmlFor="charte"
    >
      <div
        className="sc-bcXGCL dawbeU cap-checkbox"
        style={
          {
            "marginLeft": 0,
            "marginTop": 0,
          }
        }
      >
        <input
          aria-required={true}
          className="sc-bcXGCL leuqnG cap-checkbox__input"
          disabled={false}
          id="charte"
          name="charte"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={false}
          required={true}
          style={
            {
              "marginLeft": 0,
              "marginTop": 0,
            }
          }
          type="checkbox"
        />
        <div
          className="sc-bcXGCL cCxzTy cap-checkbox__box"
        />
      </div>
      <span
        className="sc-bcXGCL grQllb"
      >
        <span
          className="sc-bcXGCL sc-cc425a65-0 keKWCf hMXkYl"
        >
          registration.charte
        </span>
      </span>
    </label>
  </div>,
]
`;
